dockstore-client/src/test/java/io/dockstore/client/cli/LaunchTestIT.java

```diff
@@ -1190,6 +1198,93 @@ public void cwlMissingIdParameters() {
+        listWorkflowVersions.add(aWorkflowVersion1);
+
+        Workflow workflow = new Workflow();
+        workflow.setWorkflowVersions(listWorkflowVersions);
+        workflow.setLastModified(1);
+
+        WorkflowsApi api = mock(WorkflowsApi.class);
+        UsersApi usersApi = mock(UsersApi.class);
+        Client client = new Client();
+
+        doReturn(workflow).when(api).getPublishedWorkflowByPath(anyString(), eq(null));
+
+        WorkflowClient workflowClient = new WorkflowClient(api, usersApi, client, false);
+
+        exit.expectSystemExit();
+        exit.checkAssertionAfterwards(
+                () -> assertTrue("output should include error message",
+                        systemErrRule.getLog().contains("Cannot use workflow version 'master'")));
+
+        try {
+            workflowClient.downloadTargetEntry("quay.io/collaboratory/dockstore-tool-linux-sort", ToolDescriptor.TypeEnum.WDL, false);
+        } catch (IOException e) {
+
+        }
+    }
+
+    @Test
+    public void entry2jsonBadVersion() {
+        /*
+         * Make a runtime JSON template for input to the workflow
+         * but provide a non existent version at the end of the entry
+         * E.g dockstore workflow convert entry2json --entry quay.io/collaboratory/dockstore-tool-linux-sort:2.0.0
+         * Dockstore will try to use the last modified version (1.0.0) and print an explanation message.
+         * The last modified version is not valid so Dockstore should print an error message and exit
+         * */
+
+        WorkflowVersion aWorkflowVersion1 = new WorkflowVersion();
+        aWorkflowVersion1.setName("1.0.0");
+        aWorkflowVersion1.setValid(false);
+        Date laterDate = new Date(1000L);
+        aWorkflowVersion1.setLastModified(laterDate);
+
+        List<WorkflowVersion> listWorkflowVersions = new ArrayList<WorkflowVersion>();
+        listWorkflowVersions.add(aWorkflowVersion1);
+
+        Workflow workflow = new Workflow();
+        workflow.setWorkflowVersions(listWorkflowVersions);
+        workflow.setLastModified(1);
+
+        WorkflowsApi api = mock(WorkflowsApi.class);
+        UsersApi usersApi = mock(UsersApi.class);
+        Client client = new Client();
+
+        doReturn(workflow).when(api).getPublishedWorkflowByPath(anyString(), eq(null));
+
+        WorkflowClient workflowClient = new WorkflowClient(api, usersApi, client, false);
+
+        exit.expectSystemExit();
+        exit.checkAssertionAfterwards(
+                () -> assertTrue("output should include error messages",
+                        (systemOutRule.getLog().contains("Could not locate workflow with version '2.0.0'") &&
+                                systemErrRule.getLog().contains("Cannot use workflow version '1.0.0'"))));
+
+        try {
+            workflowClient.downloadTargetEntry("quay.io/collaboratory/dockstore-tool-linux-sort:2.0.0", ToolDescriptor.TypeEnum.WDL, false);
+        } catch (IOException e) {
+
```

Comment:
Same as previous comment on empty catch.

Reply:
Now throw IOException up the call stack so I don't have to catch it. Removed the try catch block